### PR TITLE
Set GCLOUD_PROJECT envvar during the build.

### LINKED ISSUE #340: [Runtime Builder] Symfony script fails during the build if you enable the stackdriver integration
Background:

We provide [a guide](https://docs.google.com/document/d/126NJTDNctB0zL8dwTo8-9indyWGUXpGkrXCKrHU4b-s/edit#heading=h.2djfwrz16y7j) for integrating stackdriver logging and error reporting library with symfony.

Usually a symfony project has `post-install-cmd` to run `@symfony-scripts`.

With the newest google/cloud (after v0.35), the `PsrBatchLogger` instantiate the internal connection and it needs to detect the project id.

Also the Runtime Buidler doesn't set `GCLOUD_PROJECT` envvar automatically.

Then it results that the `post-install-cmd` fails within the runtime builder, when it instantiates the PsrBatchLogger. The error message is

```
Step #1: [29-Jul-2017 07:06:21 UTC] PHP Fatal error:  Uncaught Google\Cloud\Core\Exception\GoogleException: No project ID was provided, and we were unable to detect a default project ID. in /app/vendor/google/cloud/src/Core/ClientTrait.php:173
Step #1: Stack trace:
Step #1: #0 /app/vendor/google/cloud/src/Core/ClientTrait.php(81): Google\Cloud\Logging\LoggingClient->detectProjectId(Array)
Step #1: #1 /app/vendor/google/cloud/src/Logging/LoggingClient.php(180): Google\Cloud\Logging\LoggingClient->configureAuthentication(Array)
Step #1: #2 /app/vendor/google/cloud/src/Logging/LoggingClient.php(136): Google\Cloud\Logging\LoggingClient->__construct()
Step #1: #3 /app/var/cache/prod/appProdProjectContainer.php(2070): Google\Cloud\Logging\LoggingClient::psrBatchLogger('app')
Step #1: #4 /app/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php(331): appProdProjectContainer->getPsrBatchLoggerService()
Step #1: #5 /app/var/cache/prod/appProdProjectContainer.php(2018): Symfony\Component\DependencyInjection\Container->get('psr_batch_logge...')
Step #1: #6 /app/vendor/symfony/symfony/src/Symfony/Component/DependencyIn in /app/vendor/google/cloud/src/Core/ClientTrait.php on line 173
Step #1: Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the symfony-scripts event terminated with an exception
```

A solution came to mind is to use the build-arg `BUILDER_PROJECT_ID` and set this value for an envvar `GCLOUD_PROJECT` in the builder Dockerfile:

Dockerfile
```
ARG BUILDER_PROJECT_ID
RUN env GCLOUD_PROJECT=${BUILDER_PROJECT_ID} composer.sh
```

php-latest.yaml
```
steps:
  - name: 'gcr.io/gcp-runtimes/php/gen-dockerfile:latest'
    args: ['--php71-image', 'gcr.io/google-appengine/php71:latest', '--php70-image', 'gcr.io/google-appengine/php70:latest', '--php56-image', 'gcr.io/google-appengine/php56:latest']
  - name: 'gcr.io/cloud-builders/docker:latest'
    args: ['build', '-t', '$_OUTPUT_IMAGE', '--build-arg', 'BUILDER_PROJECT_ID=$PROJECT_ID' '.']
images:
  - '$_OUTPUT_IMAGE'
```

I'll try to fix this early next week.

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #340

### CODE REVIEW

File: builder/gen-dockerfile/src/templates/Dockerfile.twig [hunk 1]
```diff
@@ -18,9 +18,11 @@ FROM {{ base_image }}
 
 {{ env_string }}
 
+ARG BUILDER_PROJECT_ID
+
 COPY . $APP_DIR
 RUN chown -R www-data.www-data $APP_DIR
-RUN /build-scripts/composer.sh
+RUN env GCLOUD_PROJECT=${BUILDER_PROJECT_ID} /build-scripts/composer.sh
```
Comment (chingor13) [inline builder/gen-dockerfile/src/templates/Dockerfile.twig]: Do we not want to just set the env variable?

Comment (tmatsuo) [inline builder/gen-dockerfile/src/templates/Dockerfile.twig]: Yeah it should be fine

**Comment by tmatsuo** [comment]
Now it uses `ENV` instruction. 

Note:

The only caveat is that it affects production runtime. At least for our managed platforms (GAE and GKE), it should be fine because our platforms override it with the correct project id. If you are using the resulted image on your own kubernetes cluster on another project, it might be problematic. You can always override it with `--env` option to the `docker run` command.

**Comment by tmatsuo** [comment]
It went to production this morning

**Pull request merged**